### ISSUE TITLE
A11y_AzureToolsStorageExplorer_BlobContainerTable_Table_ScreenReader: NVDA does not read the table cell value properly.

### ISSUE THREAD

**Issue by nirmalsahoo** [microsoft/AzureStorageExplorer#1903]
[Check out Accessibility Insights!](https://github.com/microsoft/AzureStorageExplorer/issues/url) - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

GitHubTags: #A11y_AzureToolsStorageExplorer; #A11yMAS; #A11YTCS; #DesktopApp; #MAS1.3.1;#MAS4.2.3; #NVDA; #Win32;#Benchmark;#A11ySev2;

**Environment Details:**
Desktop App: Windows Version: Windows10
NVDA Version: 2019.1.1

**Steps to Reproduce:**

1. Launch Storage Explorer
2. Navigate to "Open Connect dialog" pane and Sign in to Azure.
3. Go to "Manage Accounts" pane and Select the subscriptions you will use. Once selected, click on "Apply" button.
4. Navigate to "Blob Containers" of the selected subscription, right click and hit enter on "Create Blob container" to Create a new Blob container.
5. Once completed, Navigate to "Upload" button and Upload a blob/file by using  "Upload Files" control.
6. Navigate to Table present in right side of the storage explorer.
7. Open NVDA and check whether it reads the table and table cell properly.

**Actual:**
NVDA does not read the table properly that the NVDA focus does not go to the cell value.
Issue2:
NVDA reads the table row thrice.

Same issue is observed throughout the application for all the tables.

**Expected:**
- NVDA should read the table properly and focus should go to all cell value and read the row and column reference for particular cell.
- NVDA should read the table row only once.

Same issue is observed in JAWS also.

**Recommendations:**
Refer below link which is repository of bug fixes code snippets:
https://microsoft.sharepoint.com/teams/msenable/mas/Pages/browse-fixes.aspx

**User Impact:**
Screen reader user unable to know the table cell value in which row and column.

**MAS Reference:**
MAS1.3.1:
https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={54f28d1f-a2d1-4dcd-84e1-5c9b87e8aba4}

MAS4.2.3:
https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={30e4fb40-dcc1-45b4-b714-5e2c28b23ae8}

**Attachment for Reference**
[1903_A11y_AzureToolsStorageExplorer_BlobContainerTable_Table_Screenreader.pptx](https://github.com/microsoft/AzureStorageExplorer/files/3650665/1903_A11y_AzureToolsStorageExplorer_BlobContainerTable_Table_Screenreader.pptx)

**Comment by jinglouMSFT** [comment]
Using JAWS 2020, each table row is read only once.  The first column which is the blob name is read only once.  This is verified on window10 using JAWS.

**Comment by craxal** [comment]
Looks like it's just a problem with NVDA then. I heard that NVDA tends to me more aggressive. Marked as external

**Comment by MRayermannMSFT** [comment]
Make sure to add to VPAT and file a bug against NVDA if it's really a bug.

**Comment by craxal** [comment]
nvaccess/nvda#10548

**Comment by Alexandria-R** [comment]
This can be marked as Sev3, but not approved to close.

**Comment by jinglouMSFT** [comment]
The latest behavior using SE build 20200318.10 which includes the latest fix Craig did for the table:
- NVDA: it reads column header and the content of the cell selected of that column at the specific row. It also instructs to use the arrow key to navigate the cells.  The arrow key does work, meaning navigate through the table. 
- NVDA does read every cell three times while JAWS 2020 only reads it once
- It seems in this build of SE, except everything is read 3 times, everything else works as expected, much better than previous experienced.  
- Narrator doesn't seem to work well with Electron app. It's actually much worse than NVDA or JAWS, at least on my machine. It's really slow. 

Here is a screenshot from NVDA (note i moved my folk somewhere else in the middle of the read so it actually read out the OneNote thing:

![NVDA read](https://user-images.githubusercontent.com/10081483/77019002-a0813d00-693c-11ea-8d15-f448cbb12b92.png)

**Comment by nirmalsahoo** [comment]
@Alexandria-R Can you please take a look as it reads three times .
Please see the below screenshot.
![image](https://user-images.githubusercontent.com/53940787/77531183-85707900-6eb8-11ea-9b1c-b16dffc81e14.png)

**Comment by jinglouMSFT** [comment]
As I have pointed this out earlier, reading 3 times only occurs with NVDA.   JAWS 2020 only reads it once.  Narrator even though was very slow and hard to use, it only read once however.  Doesn't this indicate to you that this is a NVDA bug?

**Comment by Alexandria-R** [comment]
This is now considered a Sev3 issue, which means it will not block your grade C. It cannot be approved to close, however.

**Comment by ato9000** [comment]
Why is this issue being reported under Azure Cosmos DB/Shireesh Thota?

**Comment by jinglouMSFT** [comment]
@ato9000 where did you see this being reported under Cosmos DB?

**Comment by ato9000** [comment]
It is showing in Service 360: https://vnext.s360.msftcloudes.com/blades/globalandtrusted?blade=KPI:b6a943f5-486a-e3bc-8ede-b2b10c7b680c~SLA:3~Forums:All~AssignedTo:All~_loc:GlobalAndTrusted&def=&tile=severity:A11ySev2~_loc:__key__GlobalAndTrusted__b6a943f5-486a-e3bc-8ede-b2b10c7b680c&global=4:724c33bf-1ab8-4691-adb1-0e61932919c2

**Comment by jinglouMSFT** [comment]
Please email Alexandria-R to make the change. My team and I didn't set that up.  Thanks.

**Comment by Ashutoshsahoo164** [comment]
GithubTags: #Closed

**Comment by mstechie** [comment]
GitHubTags:#BM-TCS-StorageExplorer-Win32-Sep2019;
